### PR TITLE
Data Details Drawer

### PR DESCRIPTION
**Pull request by ash3rz**
I still need to fix the `Path` because the TextField for copying has some extra top padding, so it's misaligned with the label. I need to fix that in ui-lib though.

Folder details on Desktop:
image

File details on Desktop:
image

Folder details on Mobile:
image

File details on Mobile:
image
image

Permissions on Desktop:
image

Updating permissions:
image

Permissions on Mobile:
image

### CODE REVIEW

File: src/components/data/details/PermissionsPanel.js [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+                                    secondaryAction={
+                                        <>
+                                            {permissionLoadingId ===
+                                            permission.user ? (
+                                                <CircularProgress
+                                                    color="inherit"
+                                                    size={20}
+                                                />
+                                            ) : (
+                                                <PermissionSelector
```
Comment (psarando) [inline src/components/data/details/PermissionsPanel.js]: Nice! I really like this concept ✨ 

It seems this only supports 1 `permissionLoadingId` at a time, however. I can see the permission service taking a long time to respond, and in the meantime the user might want to update another user's permissions (or more than 1 in a row).

Can slow responses be simulated in storybook with `fetchMock`?

File: src/server/api/terrain.js [hunk 2]
```diff
@@ -48,24 +49,35 @@ export const handler = ({ method, pathname, headers }) => {
...
-    { method, pathname, headers, query, userID, accessToken },
+    { method, pathname, headers, query, params, userID, accessToken },
     inStream
 ) => {
     const apiURL = new URL(terrainURL);
-    apiURL.pathname = path.join(apiURL.pathname, pathname);
+    let updatedPathName = pathname;
+
+    // Replace any named path segments (:variableName) with the
+    // corresponding values
```
Comment (psarando) [inline src/server/api/terrain.js]: 👍

File: stories/data/DetailsDrawer.stories.js [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+            first_name: "Oswald",
+            source_id: "ldap",
+        },
+        freeze: {
+            email: "freeze@cyverse.org",
+            name: "Victor Fries",
+            last_name: "Fries",
+            description: "Cold",
+            id: "freeze",
+            institution: "Ice To Meet You",
```
Comment (psarando) [inline stories/data/DetailsDrawer.stories.js]: 🤣

File: stories/data/DetailsDrawer.stories.js [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+                    { user: "joker", permission: "read" },
+                    { user: "penguin", permission: "write" },
+                    { user: "freeze", permission: "read" },
+                ],
+            },
+        ],
+    };
+
+    const userInfoResp = {
+        batman: {
```
Comment (psarando) [inline stories/data/DetailsDrawer.stories.js]: :batman-approves:

File: stories/data/DetailsDrawer.stories.js [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+            },
+            {
+                id: "631334a4-52a9-11ea-b48d-008cfa5ah621",
+                value: "tag6",
+                description: "",
+            },
+        ],
+    };
+
+    const resourceTypeSelect = select(
```
Comment (psarando) [inline stories/data/DetailsDrawer.stories.js]: 👍

File: src/components/data/details/PermissionsPanel.js (hunk 1 above)
Comment (ash3rz) [reply src/components/data/details/PermissionsPanel.js]: That's an excellent point.  I'll refactor this with multiple updates in mind 👍

File: src/components/data/TagSearch.js [hunk 6]
```diff
@@ -0,0 +1,207 @@
...
+                        return option;
+                    }
+                    return option.value ? option.value : "";
+                }}
+                filterOptions={(options, params) => {
+                    // add an option which allows the user to add a new tag
+                    // when no option matches the search term
+                    const hasOption =
+                        options.findIndex((tag) => tag.value === searchTerm) !==
+                        -1;
```
[the remark this answers is not in the document]
Comment (ash3rz) [reply src/components/data/TagSearch.js]: 😄 Sometimes I get fixated on using one method for everything.  This is definitely more readable

File: src/components/data/listing/Listing.js [hunk 7]
```diff
@@ -129,6 +134,20 @@ function Listing(props) {
...
 
+    useEffect(() => {
+        callApi({
+            endpoint: `/api/filetypes/type-list`,
+        }).then(resp => {
+            if (resp) {
+                setInfoTypes(resp.types);
+            }
+        });
+    }, []);
```
[the remark this answers is not in the document]
Comment (ash3rz) [reply src/components/data/listing/Listing.js]: This sounded like a good candidate for `getStaticProps` to me, but it doesn't seem to be working. I'm probably not understanding something. I'll go ahead and merge, but write a Jira for this to get converted to caching at start up.

**Comment by ash3rz** [comment]
Thanks for the review!

**Pull request merged**